include/zephyr/net/ocpp.h

```diff
@@ -0,0 +1,257 @@
+	sa_family_t sa_family; /**< IP protocol family type 4/6 */
+};
+
+/** @brief Parameters opaque session handle for ocpp_* API */
+typedef void *ocpp_session_handle_t;
+
+/**
+ * @brief Asynchronous event notification callback registered by the
+ *        application. advised callback should not be hold for longer time
+ *        to unblock the ocpp protocol stack/lib.
+ *
+ * @param[in] reason for callback invoked.
+ * @param[in] io reffered corresponding to reason.
+ * @param[in] user_data passed on ocpp_init.
+ *
+ * @return 0 or a negative error code (errno.h)
+ */
+typedef int (*ocpp_user_notify_callback_t)(enum ocpp_notify_reason reason,
+					   union ocpp_io_value *io,
+					   void *user_data);
+
+/**
+ * @brief OCPP library init.
+ *
+ * @param[in] cpi Charge Point information
+ * @param[in] csi Central System information
+ * @param[in] cb user register callback
+ * @param[in] user_data same reference will be passed on callback
+ *
+ * @return 0 on success or a negative error code (errno.h) indicating reason of failure
+ *
+ * @note Must be called before any other ocpp API
+ */
+int ocpp_init(struct ocpp_cp_info *cpi,
+	      struct ocpp_cs_info *csi,
+	      ocpp_user_notify_callback_t cb,
+	      void *user_data);
+
+/**
+ * @brief API to request a new Session
+ *
+ * @param[out] hndl a valid opaque handle
+ *
+ * @return 0 on success or a negative error code (errno.h) indicating reason of failure
+ *
+ * @note Each connector should open unique session after ocpp_init and
+ * prior to anyother ocpp_* request message api
+ */
+int ocpp_session_open(ocpp_session_handle_t *hndl);
+
+/**
+ * @brief API to close a Session
+ *
+ * @param[in] hndl a handle received from session open
+ */
+void ocpp_session_close(ocpp_session_handle_t hndl);
+
+/**
+ * @brief Authorize request call to CS to get validity of idtag
+ *
+ * @param[in] hndl session handle
+ * @param[in] idtag (string literal) to get authorize validity
+ * @param[out] status authorization status
+ * @param[in] timeout_ms in msec
+ *
+ * @return 0 on success or a negative error code (errno.h) indicating reason of failure
+ */
+int ocpp_authorize(ocpp_session_handle_t hndl,
+		   char *idtag,
+		   enum ocpp_auth_status *status,
+		   uint32_t timeout_ms);
+
+/**
+ * @brief Notify transaction start to CS
+ *
+ * @param[in] hndl session handle
+ * @param[in] Wh energy meter reading of this connector
+ * @param[in] conn_id connector id should be > 0 and sequential number
+ * @param[in] timeout_ms timeout in msec
+ *
+ * @return: 0 on success
+ *	    EACCES - not authorized, should follow the stop charging process
+ *	    a negative error code (errno.h) indicating reason of failure
+ */
+int ocpp_start_transaction(ocpp_session_handle_t hndl,
+			   int Wh,
+			   uint8_t conn_id,
+			   uint32_t timeout_ms);
+
+/**
+ * @brief Notify transaction stopped to CS
+ *
+ * @param[in] hndl session handle
+ * @param[in] Wh energy meter reading of this connector
```

Comment:
Probably better to give a more meaningful name and not have something that starts with uppercase? And explicitly mention the unit in the description 
Fix everywhere this parameter appears